samples/OpenIdConnectSample/Startup.cs

```diff
@@ -39,6 +40,23 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory)
 
             app.Run(async context =>
             {
+                var authContext = new AuthenticateContext(CookieAuthenticationDefaults.AuthenticationScheme);
+                await context.Authentication.AuthenticateAsync(authContext);
+                if (context.Request.Path == new PathString("/remote-sign-out"))
+                {
+                    var session = authContext.Properties[OpenIdConnectSessionProperties.SessionState];
+                    var sid = context.Request.Query["sid"];
+                    if (string.Equals(sid, session, System.StringComparison.Ordinal))
```

Comment:
The specs are rather permissive as the only requirement is to include the client identifier, the origin URL and the initial state in the session state :smile: 

> **sid (session ID)**

> The sid (session ID) Claim used in ID Tokens and as a logout_uri parameter has the following definition:

> OPTIONAL. String identifier for a Session. This represents a Session of an OP at an RP to a User Agent or device for a logged-in End-User. Its contents are unique to the OP and opaque to the RP. It MUST have sufficient entropy to prevent collisions between Session IDs generated by different OPs and to prevent it from being guessed by potential attackers. Its syntax is the same as an OAuth 2.0 Client Identifier.

http://openid.net/specs/openid-connect-logout-1_0.html

> **session_state**

> Session State. JSON [RFC7159] string that represents the End-User's login state at the OP. It MUST NOT contain the space (" ") character. This value is opaque to the RP. This is REQUIRED if session management is supported.

> The Session State value is initially calculated on the server. The same Session State value is also recalculated by the OP iframe in the browser client. The generation of suitable Session State values is specified in Section 4.2, and is based on a salted cryptographic hash of Client ID, origin URL, and OP browser state. For the origin URL, the server can use the origin URL of the Authentication Response, following the algorithm specified in Section 4 of RFC 6454 [RFC6454].

> // Here, the session_state is calculated in this particular way,
// but it is entirely up to the OP how to do it under the
// requirements defined in this specification.

http://openid.net/specs/openid-connect-session-1_0.html